docs/python_based_backends.md

```diff
@@ -0,0 +1,98 @@
+<!--
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Python-based Backends
+
+Python-based backend is a special type of Triton's backends, which does
+not require any C++ code. However, this type of backends depends on
+[Python backend](https://github.com/triton-inference-server/python_backend)
+and requires the following artifacts being present:
+`libtriton_python.so`, `triton_python_backend_stub`,
+and `triton_python_backend_utils.py`.
+
+## Usage
+To implement and use a Python-based backend, make sure to follow these steps.
+* Implement the
+[`TritonPythonModel` interface](https://github.com/triton-inference-server/python_backend#usage),
+which could be re-used as a backend by multiple models.
+This script should be named `model.py`.
+* Create a folder for your custom backend under the backends directory
+(ex: /opt/tritonserver/backends) with the corresponding backend name,
+containing the `model.py`. For example, for a backend named
+`my_python_based_backend`, Triton would expect to find the full path
+`/opt/tritonserver/backends/my_python_based_backend/model.py`.
+* Make sure that `libtriton_python.so`, `triton_python_backend_stub`,
+and `triton_python_backend_utils.py` are present either under
+`/opt/tritonserver/backends/my_python_based_backend/` or
+`/opt/tritonserver/backends/python/`.
+* Specify `my_python_based_backend` as a backend in `config.pbtxt`
+for any model, that should use this backend.
```

Comment:
Would be good to mention how the priority would work in case `triton_python_backend_stub` is present in the Python based Python backend folder (i.e., this would take priority over the stub present in the Python backend folder). Additionally, please mention the use-case that the user might want to do this (i.e., use a different Python version than what is shipped by default in the container).

Reply:
Thanks for pointing this out! I'll certainly add this

Reply:
Added `Customization` section latter in the file